src/context/base.rs

```diff
@@ -9,21 +9,32 @@ use crate::media::MediaDecoder;
 use crate::node::{AudioNode, ChannelConfigOptions};
 use crate::param::AudioParamDescriptor;
 use crate::periodic_wave::{PeriodicWave, PeriodicWaveOptions};
+use crate::render::AudioProcessor;
 use crate::{node, AudioListener};
 
 /// The interface representing an audio-processing graph built from audio modules linked together,
 /// each represented by an `AudioNode`.
 ///
 /// An audio context controls both the creation of the nodes it contains and the execution of the
 /// audio processing, or decoding.
-///
-/// Please note that in rust, we need to differentiate between the [`BaseAudioContext`] trait and
-/// the [`ConcreteBaseAudioContext`] concrete implementation.
 #[allow(clippy::module_name_repetitions)]
 pub trait BaseAudioContext {
-    /// retrieves the `ConcreteBaseAudioContext` associated with this `AudioContext`
+    /// Returns the [`BaseAudioContext`] concrete type associated with this `AudioContext`
     fn base(&self) -> &ConcreteBaseAudioContext;
 
+    /// Construct a new pair of [`AudioNode`] and [`AudioProcessor`]
+    ///
+    /// The `AudioNode` lives in the user-facing control thread. The Processor is sent to the render thread.
+    fn register<
```

Comment:
Should we call this one in all `AudioNode`s contructor then? 
i.e. `context.register()` instead of `context.base().register`

Reply:
(maybe I'm wrong thinking too much as OOP...)

Reply:
yeah good point, for consistency